### PR TITLE
usertasks with security keycloak

### PR DESCRIPTION
**Pull request by nmirasch**
Integration with the keycloak security system.
I fixed the quarkus version there creating <quarkus.released.version>0.23.2</quarkus.released.version> regarding I didn't find where the snapshot is (sorry). Could you take a look, here? 
Whenever you have some time, could you take a look on this?
Any thought will be welcome, Thanks!!

### CODE REVIEW

**Comment by nmirasch** [comment]
@mswiderski I found the reason, about the keycloak extension version. It has changed . I just see the related quarkus guide has changed to  using-open-id connect. I'll try to update this PR

**Comment by mswiderski** [comment]
@nmirasch welcome to the quarkus world ... where the only constant is .. a change :) Let me know when it's ready to try it out, looking forward to it :)

**Comment by nmirasch** [comment]
@mswiderski updated!

**Comment by nmirasch** [comment]
@mswiderski I'm adding some test recreating the readme explained steps. I'll let you know when I have it added to this PR.

**Comment by nmirasch** [comment]
@mswiderski done, added some test. Ready to be tried :)

**Comment by mswiderski** [comment]
Thanks a lot. Will try it out on Monday when back home

File: kogito-usertasks-with-security-quarkus/src/main/resources/application.properties
Comment (mswiderski) [inline kogito-usertasks-with-security-quarkus/src/main/resources/application.properties]: why this change here?

File: kogito-usertasks-with-security-oidc-keycloak/src/test/java/org/acme/travels/BearerTokenAuthorizationTest.java
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/src/test/java/org/acme/travels/BearerTokenAuthorizationTest.java]: I believe this test should be disabled by default unless we have option to run some kind of embedded Keycloak...

File: kogito-usertasks-with-security-oidc-keycloak/src/main/resources/META-INF/resources/index.html
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/src/main/resources/META-INF/resources/index.html]: I guess we could remove this file completely ...

File: kogito-usertasks-with-security-oidc-keycloak/pom.xml
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/pom.xml]: this is not needed and since 0.25.0 of quarkus actually wrong we use Jackson now

File: kogito-usertasks-with-security-oidc-keycloak/pom.xml
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/pom.xml]: on master we keep 999-SNAPSHOT in general and then change the version in main pom.xml upon release

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/README.md:113]: this requires `jq` command to be present otherwise it will fail... as it was in my case :)

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/README.md]: importing of the config file fails due to duplicated roles and then it fails with messages:

`Error! Realm role 'uma_authorization' already exists.`

if I select that the import should override existing then I can this

`Error! App doesn't exist in role definitions: realm-management`

and if I skip the import of client roles then I cannot login anymore. Could you provide a bit of help on this? Maybe we import too much....

**Review by mswiderski — COMMENTED** [review]
first of all great work @nmirasch!
despite the issues mentioned I managed to make it somehow run on master realm though I think we should improve the import as it causes quite some issues. And then the export of access_token is linux specific .. so might be a good thing to have it described at least

plus would be really good to have same example with spring boot

File: kogito-usertasks-with-security-quarkus/src/main/resources/application.properties
Comment (nmirasch) [reply kogito-usertasks-with-security-quarkus/src/main/resources/application.properties]: @mswiderski the intention here was make the readme a little more consistent.. Regarding the process requires 2 managers approval and the process is checking they are not the same. Even passing 'user=john&group=managers' to the secondLineApproval would work too, we can keep it as it was :)

File: kogito-usertasks-with-security-oidc-keycloak/src/test/java/org/acme/travels/BearerTokenAuthorizationTest.java
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-keycloak/src/test/java/org/acme/travels/BearerTokenAuthorizationTest.java]: @mswiderski that's true, I tried to replicate the quarkus test way but it's true, it supposes that the keycloak is running and well configured. I thought that was something similar that we had with the infinispan service, but seeing the  https://github.com/kiegroup/kogito-runtimes/pull/197 work to ensure the dependent services running, maybe makes sense to remove this test, at least for now.

File: kogito-usertasks-with-security-oidc-keycloak/src/main/resources/META-INF/resources/index.html
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-keycloak/src/main/resources/META-INF/resources/index.html]: sure

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-keycloak/README.md:113]: @mswiderski ups, let me search for an alternative here :)

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-keycloak/README.md]: @mswiderski yes, I see what is happening.. you have to import the file after clicking 'Add Realm' button. this errors are thrown because you are trying to import the 'quarkus realm' on 'Master realm'.. Here you need to create a new one and import there the file

**Comment by mswiderski** [comment]
@nmirasch is this now ready to retest and merge?

**Comment by nmirasch** [comment]
@mswiderski,   I removed the test, changed the deps quarkus-resteasy-jsonb -> quarkus-jsonp (we need to have javax.json.JsonObject available) and quarkus snapshot version and update the readme with instructions to install the jq in different platforms. I restored the part of kogito-usertasks-with-security-quarkus only changing some urls of the readme file that had some typos. It would be great is you could give this a try.

**Comment by mswiderski** [comment]
>  quarkus-resteasy-jsonb -> quarkus-jsonp (we need to have javax.json.JsonObject available)

this is not really needed. If you upgrade quarkus version to 0.25.0 then the resteasy deps are completely redundant as quarkus-kogito is already bringing them. So just remove this dependency
https://github.com/mswiderski/kogito-quickstarts/pull/1/files#diff-2b597721e8d46e80d77ca6aa6ec490caR35-R38

Just make this last update and I will give it a try again :)

**Comment by nmirasch** [comment]
@mswiderski updated and confirmed, today it's not necessary :)
(Working fine without it)

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/README.md]: Now it is much better but I have to create the realm and then do the import with overrides. When tried to make the realm and directly import I got this error so maybe it's a matter of config of the Keycloak server

````
09:41:37,120 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (default task-3) Uncaught server error: java.lang.RuntimeException: Script upload is disabled
	at org.keycloak.keycloak-authz-policy-common@7.0.1//org.keycloak.authorization.policy.provider.js.JSPolicyProviderFactory.updatePolicy(JSPolicyProviderFactory.java:125)
	at org.keycloak.keycloak-authz-policy-common@7.0.1//org.keycloak.authorization.policy.provider.js.JSPolicyProviderFactory.onImport(JSPolicyProviderFactory.java:70)
	at org.keycloak.keycloak-server-spi-private@7.0.1//org.keycloak.models.utils.RepresentationToModel.toModel(RepresentationToModel.java:2232)
````

so maybe we can just extend the docs on describing this in two steps:
1. create new realm named quarkus
2. import config file with setting to override the existing items

wdyt?

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/README.md]: missing space after ### so it renders without making it a title

File: kogito-usertasks-with-security-oidc-keycloak/pom.xml
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-keycloak/pom.xml:34]: so in the end I was wrong ... the quarkus-jsonp is needed... not for kogito but for JWT handling.. otherwise it fails with

````
Caused by: java.lang.ClassNotFoundException: javax.json.JsonObject
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
````

so please add it back. Sorry for the trouble.

**Review by mswiderski — COMMENTED** [review]
@nmirasch this is now much better and complete - just two small comments to apply and it's ready to be merged

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-keycloak/README.md]: @mswiderski I just testing it with the docker image I usually do that in one step but anyway, in two steps with overwrite works fine.. updating the readme!

File: kogito-usertasks-with-security-oidc-keycloak/README.md
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-keycloak/README.md]: ups, updated

File: kogito-usertasks-with-security-oidc-keycloak/pom.xml
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-keycloak/pom.xml:34]: @mswiderski I swear today at my environment it works without this dep! But adding it because yesterday I believed it was 'ultra'- necessary! :S

Comment (mswiderski) [reply kogito-usertasks-with-security-oidc-keycloak/pom.xml:34]: hehe sounds like a good enough reason to me :)

**Review by mswiderski — APPROVED** [review]
so from my point of view it's ready to be merged, are you ok with that?

**Comment by nmirasch** [comment]
@mswiderski it's fine for me, Thanks!

**Comment by mswiderski** [comment]
thank you @nmirasch great work!

**Pull request merged**
